### ISSUE TITLE
Allow for fleet.yml

### ISSUE THREAD

**Issue by nickgerace** [rancher/fleet#593]
While we still want to maintain `.yaml` as our officially documented extension for `fleet.yaml`, we should add `.yml` file extension support to the code itself.

Relevant issues:
- https://github.com/rancher/fleet/issues/525
- https://github.com/rancher/fleet/issues/534
- https://github.com/rancher/fleet/issues/540
- https://github.com/rancher/fleet/issues/548

**Comment by nickgerace** [comment]
### QA Validation Steps
1. Install HA Rancher with the first Rancher RC that this issue's milestone is attached to
2. Create at least one downstream cluster (any number of nodes and any provider type)
3. Setup a public GitHub repository that uses `fleet.yml` instead of `fleet.yaml` (recommend [multi-cluster/helm](https://github.com/rancher/fleet-examples))
4. Create a GitRepo CR that uses a path to that `fleet.yml` file
5. Observe GitRepo CR and corresponding Bundle reach an `Active` state

**Comment by nickgerace** [comment]
Available on Rancher v2.6-head in Fleet v0.3.8-rc1

**Comment by davidnuzik** [comment]
**My checks PASSED**
 
**Reproduction Steps:**

Not required.

-----
 
**Validation Environment:**

Rancher version: v2.6-head `cc2b8be` 11/12/2021
Rancher cluster type: HA install
 
Downstream cluster type: n/a (not relevant)
Downstream K8s version: n/a (not relevant)
 
**Validation steps:**

1. Create two clusters - 1 node is fine for each. I used DigitalOcean node driver and default settings.
2. I forked https://github.com/rancher/fleet-examples and in my fork, modified `/mutli-cluster/helm/` such that the `fleet.yaml` file is renamed to `fleet.yml` on the master branch. This change was commited.
3. Create a GitRepo that uses a path to this (my forked) example (and thus the `fleet.yml` file)
4. Ensured the GitRepo is deployed correctly and that the Bundle appears nominal and is in an Active state.

The bundle becomes active in about 20 seconds. The repo deployment in about a minute. No issues were observed when I reviewed.
